cockroachdb/src/jepsen/cockroach/multiregister.clj

```diff
@@ -0,0 +1,131 @@
+  (->> (util/random-nonempty-subset reg-range)
+       (mapv (fn [id] [:write id (rand-int 5)]))
+       (array-map :type :invoke, :f :txn, :value)))
+
+(defrecord MultiAtomicClient [tbl-created? conn]
+  client/Client
+
+  (open! [this test node]
+    (assoc this :conn (c/client node)))
+
+  (setup! [this test]
+    ;; Everyone's gotta block until we've made the table.
+    (locking tbl-created?
+      (when (compare-and-set! tbl-created? false true)
+        (c/with-conn [c conn]
+          (info "Creating tables" (pr-str table-names))
+          (Thread/sleep 1000)
+          (doseq [t table-names]
+            (j/execute! c [(str "drop table if exists " t)]))
+          (Thread/sleep 1000)
+          (doseq [t table-names]
+            (j/execute! c [(str "create table " t
+                                " (ik int primary key, val int)")])
+            (info "Created table" t))))))
+
+  (invoke! [this test op]
+    (c/with-idempotent #{:read}
+      (c/with-exception->op op
+        (c/with-conn [c conn]
+          (c/with-timeout
+            (try
+              (c/with-txn [c c]
+                (let [[ik txn] (:value op)
+                      txn' (mapv
+                            (fn [[f id val]]
+                              (let [t (id->table id)
+                                    val' (case f
+                                           :read
+                                           (-> c
+                                               ; Look up and return current value
+                                               (c/query [(str "select val from " t " where ik = ?") ik]
+                                                        {:row-fn :val :timeout c/timeout-delay})
+                                               first)
+
+                                           :write
+                                           (do
+                                             ; Perform blind write on key, return value
+                                             (c/execute! c [(str "upsert into " t " values (?, ?)") ik val])
+                                             (cockroach/update-keyrange! test t ik)
+                                             val))]
+                                [f id val']))
+                            txn)]
+                  (assoc op :type :ok, :value (independent/tuple ik txn'))))
+              (catch org.postgresql.util.PSQLException e
+                (if (re-find #"ERROR: restart transaction" (.getMessage e))
+                  ; Definitely failed
+                  (assoc op :type :fail)
+                  (throw e)))))))))
+
+  (teardown! [this test]
+    nil)
+
+  (close! [this test]
+    (rc/close! conn)))
+
+(defn test
+  [opts]
+  (cockroach/basic-test
+   (merge
+    {:name         "multi-register"
+     :client      {:client (MultiAtomicClient. (atom false) nil)
+                   :during (independent/concurrent-generator
+                            (* 2 (count (:nodes opts)))
```

Comment:
Do you want to comment these values? They appear to configure the workload behavior but I don't know what they actually do. Maybe that's okay

Reply:
They're like this for every test (for instance, [here](https://github.com/cockroachdb/jepsen/blob/master/cockroachdb/src/jepsen/cockroach/register.clj#L93)), so I'm inclined to keep this as is.